aten/src/ATen/native/cuda/ReplicationPadding.cu

```diff
@@ -0,0 +1,239 @@
+#include "ATen/ATen.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include "ATen/cuda/CUDAContext.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/TensorUtils.h"
+#include "ATen/Utils.h"
+#include "c10/util/Exception.h"
+#include "THC/THCGeneral.h"
+#include "THC/THCNumerics.cuh"
+#include "THC/THCDeviceTensor.cuh"
+#include "THC/THCDeviceUtils.cuh"
+
+#include <algorithm>
+#include <cfloat>
+#include <cmath>
+
+
+namespace at {
+namespace native {
+__host__ __device__ __forceinline__ int imin(int a, int b) {
+      return a > b ? b : a;
+}
+
+__host__ __device__ __forceinline__ int imax(int a, int b) {
+      return a > b ? a : b;
+}
+
+__host__ __device__ __forceinline__ int iabs(int a) {
+      return a >= 0 ? a : -a;
+}
+
+namespace {
+template <typename Dtype>
+  __global__ void replication_pad(
+      const PackedTensorAccessor<Dtype, 3>& input,
+      PackedTensorAccessor<Dtype, 3>& output,
+      int padL, int padR) {
+    int outputPointId = threadIdx.x + blockIdx.x * blockDim.x;
+    int plane = blockIdx.y;
+    int batch = blockIdx.z;
+    if (outputPointId >= output.size(2)) {
+    return;
+    }
+    int outputPointX = outputPointId % output.size(2);
```

Comment:
It's not the problem we're looking for, but Is that a thing or copypasting from 2d? (see the if ... return above).